### ISSUE TITLE
Add legend to plot2d

### ISSUE THREAD

**Issue by polklin** [perrin-isir/xomx#2]
Hi,

would it be possible to consider adding a legend whenever possible when using the `plot2d` function ? 
A typical use-case would be to plot a umap embedding in which we know the associated labels for each sample (e.g. the tumor type). So in that case `adata.obs["labels"]` would be defined.

For interactive plots it is not totally necessary since we can eventually mouse over each sample (but it would still be useful). On the other hand, I would  find it nearly compulsory for plots we save (`save_dir` is defined) as we lose this interaction possibility.

Best,
Paul

**Comment by perrin-isir** [comment]
Hi Paul,
I am currently working on moving from matplotlib to Bokeh for the visualization. I will make sure to include fixed legends.

Nicolas

**Comment by perrin-isir** [comment]
Hi Paul,
I've made the modifications, now the legend on the labels should be automatically displayed. 
There are now two options for the plots, either bokeh or matplotlib, which can be chosen with `xomx.pl.extension('bokeh')` or `xomx.pl.extension('matplotlib')` (the default is bokeh).
And the plotting functions have been renamed:
- function_scatter() -> scatter() 
- function_plot() -> plot()
- plot2d() -> plot_2d_obsm()
- var_plot -> plot_var()
- plot_embedding() -> plot_2d_embedding()

There are also small modifications in the interfaces of these functions, for instance `save_dir` has been replaced by `output_file`, and violinplot is not an option anymore, it is automatically applied when relevant with matplotlib, and unavailable with bokeh. A remark about the exports (when using the `output_file` option): bokeh uses selenium and requires a webdriver to generate pngs, so it might not work straight away (in a notebook the easiest solution is to display the plot and use the interactive 'save' tool).

I will soon modify the tutorials to refect these changes.
